src/components/Carousel.tsx

```diff
@@ -1,26 +1,168 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li>
-        <img src="./img/1.png" alt="1" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="2" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="3" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="4" />
-      </li>
-    </ul>
+type Props = {
+  images: string[];
+  itemWidth?: number;
+  frameSize?: number;
+  step?: number;
+  animationDuration?: number;
+  infinite?: boolean;
+};
 
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+const Carousel: React.FC<Props> = ({
+  images,
+  itemWidth = 130,
+  frameSize = 3,
+  step = 3,
+  animationDuration = 1000,
+  infinite = false,
+}) => {
+  const [carouselList, setCarouselList] = useState(0);
+  const [stepState, setStepState] = useState(step);
+  const [frameSizeState, setFrameSizeState] = useState(frameSize);
+  const [itemWidthState, setItemWidthState] = useState(itemWidth);
+  const [animationDurationState, setAnimationDurationState] =
+    useState(animationDuration);
+  const [isInfiniteState, setIsInfiniteState] = useState(infinite);
+
+  const previous = () => {
+    if (-carouselList + stepState < 0) {
+      setCarouselList(carouselList - stepState);
```

Comment:
```suggestion
    if (carouselList - stepState < 0) {
      setCarouselList(0);
```

If the expected value is out of range just set 0 in `previous`. in `next` - the maximum value

Reply:
if I on last smile and when I press prev, it returns to the first, I think it didn't need to work so